script.js

```diff
@@ -0,0 +1,141 @@
+    let video = document.importNode(template, true);
+    video.querySelector(".ytVideo").setAttribute("id", "video" + i);
+    video.querySelector(".title").textContent = details.snippet.title;
+    video.querySelector(".views").textContent = details.statistics.viewCount + " views";
+    video.querySelector(".published").textContent = details.snippet.publishedAt.substring(0, 10).split("-").join("/");
+    video.querySelector(".author").textContent = details.snippet.channelTitle;
+    video.querySelector(".author").setAttribute("onclick", "window.open('https://www.youtube.com/channel/" + details.snippet.channelId + "')");
+    video.querySelector(".description").textContent = details.snippet.shortdescription;
+
+
+    // video.querySelector(".vid").setAttribute("src", "https://www.youtube.com/embed/" + details.id);
+    video.querySelector(".thumbnail").src = details.snippet.thumbnails.high.url;
+    video.querySelector(".thumbnail").setAttribute("onclick", "window.open('https://www.youtube.com/watch?v=" + details.id + "')");
+
+    // video.querySelector("#video" + i).setAttribute("onclick", "window.open('https://www.youtube.com/watch?v=" + details.id + "')");
+
+    document.querySelector(".main").appendChild(video);
+}
+
+
+function hideAll() {
+    for (let i = 1; i <= 15; i++) {
+        document.querySelector("#video" + i).style.display = "none";
+    }
+}
+
+
+
+function generatePages(count) {
+    let pages = document.querySelector(".pages");
+    pages.textContent = "";
+    for (let i = 1; i <= count; i++) {
+        let page = document.createElement("div");
+        page.setAttribute("class", "page");
+        let btn = document.createElement("button");
+        btn.setAttribute("id", "page" + i);
+        let pageId = i;
+        btn.setAttribute("pageId", pageId)
+        btn.setAttribute("onclick", "displayVideos(" + pageId + ")");
+        btn.textContent = i;
+        page.appendChild(btn);
+        pages.appendChild(page);
+    }
+}
+
+
+function displayVideos(pageId) {
+    hideAll();
+    let vidCount = videoCount()
+    while (Math.ceil(15 / vidCount) < pageId) {
+        pageId--;
+    }
+
+    for (let i = (vidCount * (pageId - 1)) + 1; i <= (pageId * vidCount) && i <= 15; i++) {
+        document.querySelector("#video" + i).style.display = "block";
+    }
+
+    generatePages(Math.ceil(15 / vidCount));
+    document.querySelector("#page" + pageId).classList.add("active");
+
+}
+
+function videoCount() {
+    let divWidth = document.querySelectorAll(".main")[0].offsetWidth;
+    let videoCount;
+    if (divWidth < 500) {
+        videoCount = 1;
+    } else if (divWidth < 900) {
+        videoCount = 2;
+    } else if (divWidth < 1200) {
+        videoCount = 3;
+    } else if (divWidth < 1500) {
+        videoCount = 4;
+    } else {
+        videoCount = 5;
+    }
+
+    document.querySelectorAll(".main")[0].style.setProperty('grid-template-columns', 'repeat(' + videoCount + ', 1fr)')
+    return videoCount;
```

Comment:
please help me to understand this pagination logic. Why is video count restricted to 5 numbers only?